composer/core/state.py

```diff
@@ -116,6 +118,8 @@ class State(Serializable):
     # but the getter will always return a Precision enum
     precision: Union[str, types.Precision]  # type: ignore
     _precision: types.Precision = field(init=False)  # but store an enum internally
+    steps_per_epoch: int = -1  # type: ignore
```

Comment:
Yea...it's nonstandard.

Here, I defined it as a field so you can set `steps_per_epoch` as part of the `__init__`. But it's optional since it has a default value.

`__init_` implicitly calls the property setter, which then sets the private variable `_steps_per_epoch`. The public getter `steps_per_epoch` always returns an `int` -- the actual length of the dataloader (if `_steps_per_epoch` is None), or the artificially reduced length (if the `_steps_per_epoch` is not None)

Reply:
It seems like so far there isn't a reason to set `steps_per_epoch` to the `__init__`  when constructing `State` though so what about removing it for now and then if a need arises it can be added back? Just b/c it's confusing as is